kyaml/setters2/delete_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package setters2
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func TestDelete_Filter(t *testing.T) {
+	var tests = []struct {
+		name           string
+		description    string
+		setter         string
+		input          string
+		expectedOutput string
+	}{
+		{
+			name:   "set-replicas",
+			setter: "replicas",
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  annotations:
+    replicas: 3 # {"$openapi":"replicas"}
+spec:
+  replicas: 3 # {"$openapi":"replicas"}
+ `,
+			expectedOutput: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  annotations:
+    replicas: 3
+spec:
+  replicas: 3
+ `,
+		},
+		{
+			name:        "set-foo-type",
```

Comment:
Are the names and descriptions here correct? They don't look like they have anything to do with deletion.

Reply:
Done.